types/shallowequal/index.d.ts

```diff
@@ -3,7 +3,8 @@
 // Definitions by: Sean Kelley <https://github.com/seansfkelley>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare module 'shallowequal' {
-  function shallowEqual(objA: any, objB: any, compare?: (objA: any, objB: any, indexOrKey?: number | string) => boolean, compareContext?: any): boolean;
-  export = shallowEqual;
-}
+type ShallowEqualFn = (objA: any, objB: any, compare?: (objA: any, objB: any, indexOrKey?: number | string) => boolean, compareContext?: any) => boolean;
+declare const shallowEqual: ShallowEqualFn;
+
+export = shallowEqual;
+export as namespace shallowEqual;
```

Comment:
`export as namespace` has a different meaning.. not sure if this is the intention. it means that there is a global `shallowEqual` that exists at run time. not sure if this si true for this package..

to achive what you want, you can declare teh file as:

```ts
declare function shallowEqual(objA: any, objB: any, compare?: (objA: any, objB: any, indexOrKey?: number | string) => boolean, compareContext?: any): boolean;

declare namespace shallowEqual { }

export = shallowEqual;
```

or better use `--esModuleInterop` in your project configuration if you are using TS 2.8 or later.

Reply:
Checked these types in my project and they worked. Updated the PR.

If you think this type of shimming is inappropriate for DT, then I suppose I'm okay closing the PR. I would be nice to not have to use flags to paper over compatibility issues, though.